backend-rust/src/graphql_api.rs

```diff
@@ -905,14 +905,92 @@ impl SearchResult {
-        _before: Option<String>,
+        before: Option<String>,
     ) -> ApiResult<connection::Connection<String, Block>> {
-        todo_api!()
+        let block_hash_regex: Regex = Regex::new(r"^[a-fA-F0-9]{1,64}$")
+            .map_err(|_| ApiError::InternalError("Invalid regex".to_string()))?;
+        let pool = get_pool(ctx)?;
+        let config = get_config(ctx)?;
+        let query =
+            ConnectionQuery::<i64>::new(first, after, last, before, config.block_connection_limit)?;
+        let mut connection = connection::Connection::new(false, false);
+        if !block_hash_regex.is_match(&self.query) {
+            return Ok(connection);
+        }
+        let lower_case_query = self.query.to_lowercase();
+        let rows = sqlx::query_as!(
+            Block,
+            "SELECT * FROM (
+                SELECT
+                    hash,
+                    height,
+                    slot_time,
+                    block_time,
+                    finalization_time,
+                    baker_id,
+                    total_amount
+                FROM blocks
+                WHERE
+                    height = $5
+                    OR starts_with(hash, $6)
+                    AND height > $1
+                    AND height < $2
+                ORDER BY
+                    (CASE WHEN $4 THEN height END) ASC,
+                    (CASE WHEN NOT $4 THEN height END) DESC
+                LIMIT $3
+            ) ORDER BY height DESC",
+            query.from,
+            query.to,
+            query.limit,
+            query.desc,
+            lower_case_query.parse::<i64>().ok(),
+            lower_case_query
+        )
+        .fetch_all(pool)
+        .await?;
+
+        let mut min_height = None;
+        let mut max_height = None;
+        for block in rows {
+            min_height = Some(match min_height {
+                None => block.height,
+                Some(current_min) => min(current_min, block.height),
+            });
+
+            max_height = Some(match max_height {
+                None => block.height,
+                Some(current_max) => max(current_max, block.height),
+            });
+            connection.edges.push(connection::Edge::new(block.height.to_string(), block));
+        }
+
+        if let (Some(page_min_height), Some(page_max_height)) = (min_height, max_height) {
+            let result = sqlx::query!(
+                r#"
+                    SELECT MAX(height) as max_height, MIN(height) as min_height
+                    FROM blocks
+                    WHERE
+                        height = $1
+                        OR starts_with(hash, $2)
+                "#,
+                lower_case_query.parse::<i64>().ok(),
+                lower_case_query,
+            )
+            .fetch_one(pool)
+            .await?;
+            connection.has_previous_page =
+                result.max_height.map_or(false, |db_max| db_max > page_max_height);
```

Comment:
We are moving from big to small here. 

This means that we have to reverse it all